docs(install): update deployment_guide to include AWS EKS Addon details (#1504)
accuknox/KubeArmor: getting-started/deployment_guide.md

You can find more details about helm related values and configurations [here](https://github.com/kubearmor/KubeArmor/tree/main/deployments/helm/KubeArmorOperator).

<details>
  <summary><h4>EKS-Plugin installation</h4></summary>
  

> Here, you will understand how to install KubeArmor on your EKS cluster using [Amazon EKS add-on](https://docs.aws.amazon.com/eks/latest/userguide/eks-add-ons.html). 

Reviewer: ```suggestion
If you are using AWS EKS, You can use [Amazon EKS add-on](https://docs.aws.amazon.com/eks/latest/userguide/eks-add-ons.html) to install KubeArmor in your AWS EKS Cluster. 
```

Please improvise as needed

Author: Resolved

> ### From AWS Console
> 
> Once you've subscribed to Kubearmor on AWS Marketplace and completed the on-screen setup, you'll be directed to the Amazon EKS console. To begin in the Amazon EKS console, navigate to your EKS clusters and click on the "Add-ons" tab. From there, select "Get more add-ons" to locate the KubeArmor EKS add-ons in the cluster settings of your existing EKS clusters. You can use the search bar to find "KubeArmor" and then follow the on-screen instructions to enable the Kubearmor add-on for your Amazon EKS cluster.
> 
> ### From AWS-CLI

Reviewer: ```suggestion
**From AWS Console**

Once you've subscribed to Kubearmor on AWS Marketplace and completed the on-screen setup, you'll be directed to the Amazon EKS console. To begin in the Amazon EKS console, navigate to your EKS clusters and click on the "Add-ons" tab. From there, select "Get more add-ons" to locate the KubeArmor EKS add-ons in the cluster settings of your existing EKS clusters. You can use the search bar to find "KubeArmor" and then follow the on-screen instructions to enable the Kubearmor add-on for your Amazon EKS cluster.

**From AWS-CLI**
```

Main heading is h4, so let's move these headings to h4 or bold (h5 doesn't exist)

Author: Resolved

> [!NOTE] 
> kArmor CLI provides a Developer Friendly way to interact with KubeArmor Telemetry. You can stream KubeArmor telemetry independently of kArmor CLI tool and integrate it with your chosen SIEM (Security Information and Event Management) solutions. [Here's a guide](https://github.com/kubearmor/kubearmor-relay-server/blob/main/README.md#streaming-kubearmor-telemetry-to-external-siem-tools) on how to achieve this integration. This guide assumes you have kArmor CLI to access KubeArmor Telemetry but you can view it on your SIEM tool once integrated.

> ## KubeArmor EKS-plugin 

Reviewer: Can we keep this heading entirely as a collapsible section under install. Next line to helm install.

Author: Created a collapsible section for Eks-plugin installation under install .

> > Amazon Elastic Kubernetes Service (Amazon EKS) is a managed Kubernetes service provided by Amazon Web Services (AWS) which simplifies the process of running Kubernetes clusters by handling the underlying infrastructure and operational aspects. KubeArmor’s integration with EKS can provide a user with a tool that can enforce security policies for them at the pod level as well as at the node level. Users can restrict all events related to network, process, and file using these policies.

Reviewer: I don't think we need to explain EKS and KubeArmor here. It's a KubeArmor deployment guide so people already know about Kubernetes and Kubearmor here. Linking to the description of EKS in a hyperlink would be enough. WDYT?

Author: Removed unnecessary information and hyperlinked EKS-plugin official documentation.
